### DISCUSSION TITLE
Bravo showing incorrect dates in date table.

### DISCUSSION THREAD

**Discussion by marksmith007** [sql-bi/Bravo#847] [Q&A]
The dates used to be good in my model but now showing incorrect years:
Date
7/4/241
7/11/241

etc, I have tried rescanning etc. A few weeks ago i moved from MAC (running emulation) to Windows and had to install Bravo again but at the point, from what I recall everything looked good. I have the latest version installed.

The date table still seems to work, its just the date table with the strange format

Very odd?

**Comment by albertospelta** [comment]
@marksmith007 Just to clarify, where exactly are you seeing the dates in this strange format? Is it in Bravo or in Power BI Desktop? If you could share a screenshot, that’d really help us understand better.

Also, what language is your Windows set to? If possible, could you send us a .pbix file that shows the issue? It doesn’t need to be your full model - just a simple one with the date table created by Bravo that reproduces the problem.

**Reply by marksmith007** [reply]
<img width="939" alt="Capture2" src="https://github.com/user-attachments/assets/0e1135f3-c5cf-47cb-a3c8-e935c6ac7e15">
<img width="750" alt="Capture1" src="https://github.com/user-attachments/assets/57b27d87-e5a0-4686-8621-58a2467724f1">
Its showing both the the tool and t=in the Date Table in Desktop.

**Reply by albertospelta** [reply]
This could be related to the range of dates in your model. To verify, you can: 

- Copy the code of the variables `__FirstYear` and `__LastYear` from the calculated table `DateAutoTemplate`

![image](https://github.com/user-attachments/assets/d4a0e0bc-81e7-43d4-a9e2-f1f123031782)

- Replace the code in the query below and run the query on your model to verify that the values represent a valid range of years
```
EVALUATE  
VAR __FirstYear = YEAR ( MINX ( { MIN ( 'Customer'[Birth Date] ), MIN ( 'Sales'[Due Date] ) }, ''[Value] ) )  
VAR __LastYear =  YEAR ( MAXX ( { MAX ( 'Customer'[Birth Date] ), MAX ( 'Sales'[Due Date] ) }, ''[Value] ) )  
RETURN { __FirstYear, __LastYear }  
```

**Reply by marksmith007** [reply]
Thanks I will give is a try on Monday
